azure/scope/cluster.go

```diff
@@ -462,6 +464,20 @@ func (s *ClusterScope) SetNatGatewayIDInSubnets(name string, id string) {
 	}
 }
 
+// UpdateSubnetCIDRs updates the subnet CIDRs for the subnet with the same name.
+func (s *ClusterScope) UpdateSubnetCIDRs(name string, cidrBlocks []string) {
```

Comment:
This isn't a complex function, but would it still be worth adding a unit test for it in `cluster_test.go`?

Same question for `UpdateSubnetID` below.

Reply:
It seems like we don't have unit tests for the other getter/setter functions like `Subnet()` or `SetSubnet()`. In that case, do you think we should add tests for all of them? Or just leave it be for now?

Reply:
I think we can just leave it be then. 👍🏻